packages/web-lib/hooks/useBalances.tsx

```diff
@@ -27,88 +27,124 @@ const		defaultStatus = {
-	const getBalances = useCallback(async (): Promise<{[key: string]: Types.TBalanceData}> => {
+	const getBalances = useCallback(async (): Promise<void> => {
 		if (!isActive || !web3Address || (props?.tokens || []).length === 0) {
-			return {};
+			return;
 		}
 
-		set_status({...defaultStatus, isLoading: true, isFetching: true, isRefetching: defaultStatus.isFetched ? true : false});
+		set_status({
+			...defaultStatus,
+			isLoading: true,
+			isFetching: true,
+			isRefetching: defaultStatus.isFetched ? true : false
+		});
+
 		let		currentProvider = providers.getProvider(props?.chainID || web3ChainID || 1);
 		if (props?.chainID === web3ChainID && provider) {
 			currentProvider = provider as ethers.providers.BaseProvider | ethers.providers.Web3Provider;
 		}
 
-		const	{lensAddress} = networks[props?.chainID || web3ChainID];
-		let		lensContract: Contract | undefined = undefined;
-		if (lensAddress) {
-			lensContract = new Contract(lensAddress, LENS_ABI);
-		}
-
 		const	calls = [];
 		const	ethcallProvider = await providers.newEthCallProvider(currentProvider);
 		for (const element of (props?.tokens || [])) {
-			const	token = element.token as string;
+			const	{token} = element;
 			const	ownerAddress = (element?.for || web3Address) as string;
-			const	tokenContract = new Contract(token, ERC20_ABI);
-
-			calls.push(...[
-				tokenContract.balanceOf(ownerAddress),
-				tokenContract.decimals(),
-				tokenContract.symbol(),
-				lensContract ? lensContract.getPriceUsdcRecommended(token) : tokenContract.balanceOf(ownerAddress)
-			]);
+			const	isEth = toAddress(token) === ETH_TOKEN_ADDRESS;
+			if (isEth) {
+				const	tokenContract = new Contract(WETH_TOKEN_ADDRESS, ERC20_ABI);
+				calls.push(
+					ethcallProvider.getEthBalance(ownerAddress),
+					tokenContract.decimals(),
+					tokenContract.symbol()
+				);
+			} else {
+				const	tokenContract = new Contract(token, ERC20_ABI);
+				calls.push(
+					tokenContract.balanceOf(ownerAddress),
+					tokenContract.decimals(),
+					tokenContract.symbol()
+				);
+			}
 		}
 
 		const	_data: {[key: string]: Types.TBalanceData} = {};
 		try {
 			const	results = await ethcallProvider.tryAll(calls);
 			let		rIndex = 0;
 			for (const element of (props?.tokens || [])) {
-				const	token = element.token as string;
+				const	{token} = element;
 				const	balanceOf = results[rIndex++] as BigNumber;
 				const	decimals = results[rIndex++] as number;
-				const	symbol = results[rIndex++] as string;
-				const	price = results[rIndex++] as BigNumber;
+				const	rawPrice = format.BN(props?.prices?.[toAddress(token)] || ethers.constants.Zero);
+				let symbol = results[rIndex++] as string;
+
+				if (toAddress(token) === ETH_TOKEN_ADDRESS) {
+					symbol = 'ETH';
+				}
 				_data[toAddress(token)] = {
 					decimals: Number(decimals),
 					symbol: symbol,
 					raw: balanceOf,
-					rawPrice: lensContract ? price : ethers.constants.Zero,
+					rawPrice: rawPrice,
```

Comment:
nit of nits: Then we can just
```suggestion
					rawPrice,
```